packages/grafana-ui/src/components/Select/CustomInput.tsx

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { components, InputProps } from 'react-select';
+
+/**
+ * Custom input component for react-select to add data-testid attribute
+ */
+export const CustomInput = (props: InputProps) => {
+  let testId;
+
+  if ('data-testid' in props.selectProps) {
```

Comment:
looking at the failing test, do we need an extra check here?
maybe 
```suggestion
  if ('data-testid' in props.selectProps && props.selectProps['data-testid']) {
```

Reply:
Good catch!